### ISSUE TITLE
Investigate how to create a megalinter plugin

### ISSUE THREAD

**Issue by Ellerbach** [Ellerbach/docfx-companion-tools#30]
See https://megalinter.io/latest/plugins/
Megalinter allows to create plugin. This can be a nice and convenient way to get the tools integrated in a larger linter infrastructure.

**Comment by nvuillam** [comment]
Great idea :)
If you need support, don't hesitate to ask in the repo issues :)

**Comment by Ellerbach** [comment]
@nvuillam feel free to help :-) PR very welcome as well ;-)

**Comment by nvuillam** [comment]
@Ellerbach it is basically one YML descriptor file with some info about how to call our linter, + one (optional but recommended) README :)

Example: 
- descriptor: https://github.com/one-acre-fund/mega-linter-plugin-logstash/blob/main/mega-linter-plugin-logstash/logstash.megalinter-descriptor.yml
- doc: https://github.com/one-acre-fund/mega-linter-plugin-logstash

Much easier to write for people with the knowledge about the linters, so with my current backlog I'm afraid the PR won't come from me ^^

**Comment by Ellerbach** [comment]
Thanks, we'll definitely look at this. We'll ping you here/open issue if we have questions.

**Comment by DariuszPorowski** [comment]
Hey @nvuillam :) I have a few Q? :)

- How to run plugin together with documentation flavor?
- How to use GITHUB_TOKEN passed to mega-linter for tool from plugin's descriptor?
- How to use debian/ubuntu install style for tool's installation in plugin descriptor? know only `RUN` is possible but it looks like only alpine setup is supported, or I missed something?

**Comment by nvuillam** [comment]
@DariuszPorowski 

Like any other plugin , using PLUGINS property :) -> https://megalinter.io/beta/plugins/#use-external-plugins
If docfx is dotnet based as I suspect, you should better use dotnet flavor to avoid to time required to install dotnet on documentation flavor ^^

With secured variables, it's not passed anymore -> https://megalinter.io/beta/configuration/#environment-variables-security
I could create a SECURED_VARIABLES_ALLOWED=GITHUB_TOKEN , meanwhile you can override SECURED_ENV_VARIABLES_DEFAULT

No debian/ubuntu, as it just runs the command on MegaLinter alpine-based image before running the linters :)

**Comment by DariuszPorowski** [comment]
@nvuillam will look at vars, but Q around flavor was because I created the configuration exactly in that matter, but when I use docs flavor, it shows me no litter found. And it's not error from the litter container but ML config :D on full image all is good

**Comment by nvuillam** [comment]
There was a bug using plugins with flavors, but i think we solved it, did you try with beta version?

**Comment by DariuszPorowski** [comment]
And one more, pulling GH action with ML takes 2-3 mins each time. Have you explored any options to cache it inside gh actions?

**Comment by DariuszPorowski** [comment]
> There was a bug using plugins with flavors, but i think we solved it, did you try with beta version?

Hop, the latest stable one. Will try with beta. :)

**Comment by nvuillam** [comment]
> And one more, pulling GH action with ML takes 2-3 mins each time. Have you explored any options to cache it inside gh actions?

I think @Kurt-Von-Laven tried it... me not personally yet ^^

If you have a simple recipe i'm all ears :)

**Comment by Kurt-von-Laven** [comment]
Yes, we created [ScribeMD/docker-cache](https://github.com/ScribeMD/docker-cache) for this purpose. If you need still better performance, one could also add an argument to mega-linter-runner so that it pulls from ghcr.io by default in v7 but can still pull from Docker Hub.

**Comment by DariuszPorowski** [comment]
@Kurt-von-Laven will try, thanks!

**Comment by DariuszPorowski** [comment]
@Kurt-von-Laven @nvuillam tested, [ScribeMD/docker-cache](https://github.com/ScribeMD/docker-cache) works, does the job, had to switch from official ml action to npx mega-linter-runner, but it looks like restore from cache is longer than pulling each time. pulling takes on avg ~35s, restoring from cache ~50s (documentation flavor). So probably for bigger images like full megalinter makes sens :)

**Comment by Kurt-von-Laven** [comment]
Shoot. docker-cache does have some overhead, because it has to extract and load the image from the cache rather than pull layers directly. I am glad you measured! Your best bet is probably to switch mega-linter-runner to use ghcr.io instead of Docker Hub then since that will get you the best pull times. Which runner did you test on for my awareness?

**Comment by DariuszPorowski** [comment]
@Kurt-von-Laven used ubuntu-latest, will try with ghcr.io, and let you know :)
